keyCalculation_test.go

```diff
@@ -0,0 +1,57 @@
+package dynsampler
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+)
+
+func randomFieldName() string {
+	dictionary := []string{
+		"http.response.status_code",
+		"http.request.method",
+		"http.request.route",
+		"service.name",
+	}
+	return dictionary[rand.Intn(len(dictionary))]
+}
+
+// generateTestData creates a map of test data with the specified number of keys
+func generateTestData(numKeys int) map[string]float64 {
```

Comment:
this probably isn't a good simulator for things like the field names above, but eh, it's a start